examples/jsm/exporters/GLTFExporter.js

```diff
@@ -2962,6 +2969,67 @@ class GLTFMaterialsEmissiveStrengthExtension {
 
 }
 
+/**
+ * GPU Instancing Extension
+ *
+ * Specification: https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Vendor/EXT_mesh_gpu_instancing
+ */
+class GLTFMeshGpuInstancing {
+
+	constructor( writer ) {
+
+		this.writer = writer;
+		this.name = 'EXT_mesh_gpu_instancing';
+
+	}
+
+	writeNode( object, nodeDef ) {
+
+		if ( ! object.isInstancedMesh ) return;
+
+		const writer = this.writer;
+
+		const mesh = object;
+
+		const translationAttr = new Float32Array( mesh.count * 3 );
+		const rotationAttr = new Float32Array( mesh.count * 4 );
+		const scaleAttr = new Float32Array( mesh.count * 3 );
+
+		const matrix = new Matrix4();
+		const position = new Vector3();
+		const quaternion = new Quaternion();
+		const scale = new Vector3();
+
+		for ( let i = 0; i < mesh.count; i ++ ) {
+
+			mesh.getMatrixAt( i, matrix );
+			matrix.decompose( position, quaternion, scale );
+
+			position.toArray( translationAttr, i * 3 );
+			quaternion.toArray( rotationAttr, i * 4 );
+			scale.toArray( scaleAttr, i * 3 );
+
+		}
+
+		const attributes = {
+			TRANSLATION: writer.processAccessor( new BufferAttribute( translationAttr, 3 ) ),
+			ROTATION: writer.processAccessor( new BufferAttribute( rotationAttr, 4 ) ),
+			SCALE: writer.processAccessor( new BufferAttribute( scaleAttr, 3 ) ),
+		};
+
+		if ( mesh.instanceColor )
+			attributes.COLOR = writer.processAccessor( mesh.instanceColor );
```

Comment:
Because `COLOR` is not an attribute name in either the glTF or `EXT_mesh_gpu_instancing` specifications, we probably shouldn't use it here — non-standard semantics should be prefixed with an `_`.

We could either follow the vertex color naming convention, `COLOR_0`, or choose our own name prefixed with an underscoree, like `_COLOR`. I'm leaning somewhat toward `COLOR_0`.

Reply:
Okay since the spec suggests prefixing underscore for custom instance attribute, we can use `_COLOR`, or maybe `_COLOR_0` ?

`COLOR_0` is also fine, but should that be proposed in the spec as well, since it's not prefixed?

Reply:
Let's do `_COLOR_0` to be on the safe side here, and can update it if the spec moves that direction.

Reply:
Done